src/theory/uf/theory_uf_rewriter.cpp

```diff
@@ -236,6 +236,27 @@ Node TheoryUfRewriter::rewriteLambda(Node node)
   {
     return retElimShadow;
   }
+  // see if it can be eliminated, (lambda ((x T)) (f x)) ---> f
+  if (node[1].getKind() == Kind::APPLY_UF)
+  {
+    size_t nvar = node[0].getNumChildren();
+    if (node[1].getNumChildren() == nvar)
+    {
+      bool matchesList = true;
+      for (size_t i = 0; i < nvar; i++)
+      {
+        if (node[0][i] != node[1][i])
+        {
+          matchesList = false;
+          break;
+        }
+      }
```

Comment:
Are node[0] and node[1] hash-consed? If yes, would it be enough to compare `node[0] == node[1]`?

Reply:
No, they have different kind (APPLY_UF vs BOUND_VAR_LIST).